variant/tokenizers/insertion_base.py

```diff
@@ -0,0 +1,97 @@
+"""A module for Insertion Tokenization Base Class."""
+import re
+from abc import abstractmethod
+from typing import Optional, Dict
+from .tokenizer import Tokenizer
+from variant.schemas.token_response_schema import Insertion, TokenMatchType
+from .caches import AminoAcidCache, NucleotideCache
+from .tokenize_base import TokenizeBase
+
+
+class InsertionBase(Tokenizer):
+    """Class for tokenizing Insertion."""
+
+    def __init__(self, amino_acid_cache: AminoAcidCache,
+                 nucleotide_cache: NucleotideCache) -> None:
+        """Initialize the Insertion Class.
+
+        :param AminoAcidCache amino_acid_cache: Valid amino acid codes
+        :param NucleotideCache nucleotide_cache: Valid nucleotides
+        """
+        self.splitter = re.compile('ins')
```

Comment:
fwiw, depending on string size, str.split() is sometimes a little more performant when your regex is just a literal character sequence w/o any wildcards

```
In [3]: token = 'g.123_124insAGC'

In [4]: %timeit token.split('ins')
136 ns ± 1.31 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [6]: import re

In [7]: splitter = re.compile('ins')

In [8]: %timeit splitter.split(token)
250 ns ± 6.14 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Reply:
Okay! I think this would be a separate issue to focus on improving performance.